File: 1951-find-the-winner-of-the-circular-game/find-the-winner-of-the-circular-game.cpp
class Solution {
public:
    int findTheWinner(int n, int k) {
    vector<int> p(n, 0);
        int currPos = 0;
        int ones = 0;

        while (ones < n - 1) {
            int c = 0;
            while (c < k) {
                currPos = (currPos + n) % n;
                if (p[currPos] == 0)
                    c++;
                currPos++;
            }
            currPos = (currPos + n - 1) % n;
            p[currPos] = 1;
            ones++;
            while (p[currPos] == 1) {
                currPos++;
                currPos = (currPos + n) % n;
            }
        }

        int ans = 0;
        for (int i = 0; i < n; i++) {
            if (p[i] == 0) {
                ans = i + 1;
                break;
            }
        }
        return ans;
         }

};

File: 1951-find-the-winner-of-the-circular-game/find-the-winner-of-the-circular-game.java
class Solution {
    public int findTheWinner(int n, int k) {
        List<Integer> friends = new LinkedList<Integer>();
        for (int i = 1; i <= n; i++) {
            friends.add(i);
        }

        int i=0;
        while (friends.size()!=1){
            int  killedFriendPos = (i+k-1) % friends.size();
            friends.remove(killedFriendPos);
            i =killedFriendPos ;
        }

        return friends.get(0);
    }
}